From 7c6edadd35a43cba9d813885facaead7d8b5b518 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 23 Jun 2019 11:57:01 +0300
Subject: [PATCH] [SW-0] uart: add preboot_puts_hex_raw() function

The new preboot_puts_hex_raw() function is printing the provided string
and afterwards the unsigned integer value in hexadecimal base format.
The EOL is not printed, as the function name suggests ("raw").

Change-Id: Ieb2f234d64afda6d951ea578277742167f978efd
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/uart.c b/common/uart.c
index 20c38c6..f08af2b 100644
--- a/common/uart.c
+++ b/common/uart.c
@@ -83,6 +83,13 @@
 	preboot_puts("\n");
 }
 
+void preboot_puts_hex_raw(const char *log, unsigned int val, bool pad)
+{
+	preboot_puts_raw(log);
+	preboot_puts_raw("0x");
+	preboot_puth_raw(val, pad);
+}
+
 void preboot_puts_hex(const char *log, unsigned int val, bool pad)
 {
 	preboot_puts_raw(log);
diff --git a/common/uart.h b/common/uart.h
index 595ff10..7627a3e 100644
--- a/common/uart.h
+++ b/common/uart.h
@@ -17,6 +17,7 @@
 int preboot_uint2hexstring(char *str, unsigned int num, bool pad);
 void preboot_puth_raw(unsigned int val, bool pad);
 void preboot_puth(unsigned int val, bool pad);
+void preboot_puts_hex_raw(const char *log, unsigned int val, bool pad);
 void preboot_puts_hex(const char *log, unsigned int val, bool pad);
 void preboot_puts_char(const char *str, unsigned char val);
 
